### PR TITLE
SIM-2896: Update the docstrings for most demos.

### PR DESCRIPTION
**Pull request by fablechris**
Add a follow/focus to the demos where it makes sense. 
Fix a validation bug that caused an infinite loop if no validator was provided.

### CODE REVIEW

File: python/demos/character_commands.py [hunk 1]
```diff
@@ -10,16 +10,25 @@ class EnableAgentDemo(Demo):
...
             function=self.enable_agent_demo,
         )
 
     def enable_agent_demo(self, bridge: RuntimeBridge):
         """
-        Enable or disable an agent
-
-        :param bridge: The bridge to the runtime.
+        On simulation start, the chosen character's bridge agent is enabled or disabled. Enabled agents use the
+        Bridge to generate their actions and will contribute to the overall cost of running the simulation. Disabled
```
Comment (frankcarey) [inline python/demos/character_commands.py:21]: .. will contribute to the overall cost of running the simulation [when using a paid LLM service like OpenAI.]

File: python/demos/character_commands.py [hunk 2]
```diff
@@ -10,16 +10,25 @@ class EnableAgentDemo(Demo):
...
+        agents choose their actions directly in the simulation Runtime using a simple scoring system instead of sending
```
Comment (frankcarey) [inline python/demos/character_commands.py:22]: .. simpler utility AI based on scored affordances (like the SIMs).

**Review by frankcarey — CHANGES_REQUESTED** [review]
just a couple notes, and we can merge

**Review by DWGameDev — COMMENTED** [review]
Tested and encountered no new errors.

**Pull request merged**